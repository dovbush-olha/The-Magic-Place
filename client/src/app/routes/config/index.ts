export { ROUTES } from './routes.ts';
export type { RouteParams } from './route-params.ts';
export { routeParams } from './route-params.ts';
