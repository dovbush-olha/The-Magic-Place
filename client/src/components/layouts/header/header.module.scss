@import "../../../scss/settings/colors";

.header {
  position: sticky;
  offset: 0;
  background-color: color(background, dark-blue);
}

.navbar {
  height: 9.25rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  margin: 0 auto;
  max-width: 72.5rem;
  padding-inline: 32px;
}

.navLeft,
.navRight {
  flex-grow: 1;
  display: flex;
}

.navLeft {
  justify-content: flex-start;
}

.navRight {
  justify-content: flex-end;
}

.logo {
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
}
