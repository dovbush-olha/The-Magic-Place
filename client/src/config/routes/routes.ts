import type { RouteParams } from './route-params.ts';

export const ROUTES = {
  homeRoute: () => '/',
  settingsRoute: () => '/settings',
  profileRoute: () => '/profile',
  allSpellsRoutes: () => '/spells',
  viewSpellRoute: ({ spellName }: RouteParams['viewSpellParams']) => `/spells/${spellName}`,
};
