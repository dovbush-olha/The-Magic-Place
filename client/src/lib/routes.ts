export const ROUTES = {
  getAllSpells: () => '/spells',
  getSpell: ({ spellName }: { spellName: string }) => `/spells/${spellName}`,
};
