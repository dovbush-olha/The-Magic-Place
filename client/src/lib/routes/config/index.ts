export { ROUTES } from './routes.ts';
export type { RouteParams } from './routeParams.ts';
export { routeParams } from './routeParams.ts';
