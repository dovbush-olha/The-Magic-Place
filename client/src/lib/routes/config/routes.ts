import type { RouteParams } from './route-params.ts';

export const ROUTES = {
  homeRoute: () => '/',
  allSpellsRoutes: () => '/spells',
  viewSpellRoute: ({ spellName }: RouteParams['viewSpellParams']) => `/spells/${spellName}`,
};
