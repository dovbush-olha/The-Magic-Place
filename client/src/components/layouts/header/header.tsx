import Logo from '@/assets/logo-icon.png';
import { NavLink } from '@/components/navigation/nav-link';
import { ROUTES } from '@/config/routes';

import styles from './header.module.scss';

export function Header() {
  return (
    <header className={styles.header}>
      <nav className={styles.navbar}>
        <div className={styles.navLeft}>
          <NavLink to={ROUTES.homeRoute()}>Home</NavLink>
        </div>

        <div className={styles.logo}>
          <img
            src={Logo}
            alt="logo"
            height={100}
            width={100}
          />
        </div>

        <div className={styles.navRight}>
          <NavLink to={ROUTES.profileRoute()}>Profile</NavLink>
          <NavLink to={ROUTES.settingsRoute()}>Settings</NavLink>
        </div>
      </nav>
    </header>
  );
}
