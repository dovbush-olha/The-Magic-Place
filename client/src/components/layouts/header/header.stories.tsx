import { Meta, StoryObj } from '@storybook/react';

import { Header } from './header';

const meta: Meta<typeof Header> = {
  title: 'Layouts/Header',
  component: Header,
};

export default meta;

type Story = StoryObj<typeof Header>;

export const Default: Story = {
  args: {},
};
