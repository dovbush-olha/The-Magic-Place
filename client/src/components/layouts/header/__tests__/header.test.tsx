import { ROUTES } from '@/config/routes';
import { renderApp, screen } from '@/testing/test-utils.tsx';

import { Header } from '../header.jsx';

describe('Header component', () => {
  it('matches snapshot', async () => {
    const { asFragment } = await renderApp(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders navigation links with correct text and paths', async () => {
    await renderApp(<Header />);

    const homeLink = screen.getByRole('link', { name: /home/i });
    expect(homeLink).toBeInTheDocument();
    expect(homeLink).toHaveAttribute('href', ROUTES.homeRoute());

    const profileLink = screen.getByRole('link', { name: /profile/i });
    expect(profileLink).toBeInTheDocument();
    expect(profileLink).toHaveAttribute('href', ROUTES.profileRoute());

    const settingsLink = screen.getByRole('link', { name: /settings/i });
    expect(settingsLink).toBeInTheDocument();
    expect(settingsLink).toHaveAttribute('href', ROUTES.settingsRoute());
  });
});
