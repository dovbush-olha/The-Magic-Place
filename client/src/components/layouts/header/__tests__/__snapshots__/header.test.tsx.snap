// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="_header_31cdbd"
  >
    <nav
      class="_navbar_31cdbd"
    >
      <div
        class="_navLeft_31cdbd"
      >
        <a
          aria-current="page"
          class="_navLink_101612 active"
          href="/"
        >
          Home
        </a>
      </div>
      <div
        class="_logo_31cdbd"
      >
        <img
          alt="logo"
          height="100"
          src="/src/assets/logo-icon.png"
          width="100"
        />
      </div>
      <div
        class="_navRight_31cdbd"
      >
        <a
          class="_navLink_101612"
          href="/profile"
        >
          Profile
        </a>
        <a
          class="_navLink_101612"
          href="/settings"
        >
          Settings
        </a>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
